### PR TITLE
TableView CSF 3 update for FocusRing clipping

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/3825

Updated to CSF 3 while I was at it so I could test more combinations of density and hideHeader and not add yet another story.

~This does not address height yet. Will need to consult design.~

UPDATE: Consulted design. For now the top/bottom cut off is acceptable. It'll be addressed when Buttons support t-shirt sizing.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to story hideHeader, and focus a button inside the hideHeader column.

### CODE REVIEW

File: packages/@react-spectrum/table/stories/Table.stories.tsx [hunk 1]
```diff
@@ -71,1531 +170,739 @@ let items = [
...
 
-let columnsSomeFR: PokemonColumn[] = [
-  {name: 'Name', uid: 'name', width: '1fr'},
-  {name: 'Type', uid: 'type', width: '1fr'},
-  {name: 'Height', uid: 'height'},
-  {name: 'Weight', uid: 'weight'},
-  {name: 'Level', uid: 'level', width: '4fr'}
-];
 
 function AsyncLoadingExample(props) {
```
Comment (reidbarber) [inline packages/@react-spectrum/table/stories/Table.stories.tsx:907]: Maybe a good place to sneak this in?

```
let url = new URL('https://www.reddit.com/r/upliftingnews.json');
```

File: packages/@react-spectrum/table/stories/Table.stories.tsx [hunk 2]
```diff
@@ -2041,3 +1702,90 @@ function ZoomResizing() {
...
+  {name: 'Type', uid: 'type'},
+  {name: 'Height', uid: 'height'},
+  {name: 'Weight', uid: 'weight'},
+  {name: 'Level', uid: 'level'}
+];
+
+export const ResizingControlledNoInitialWidths: TableStory = {
+  render: () => (
+    <ControllingResize width={900} columns={uncontrolledColumns} />
+  ),
```
Comment (reidbarber) [inline packages/@react-spectrum/table/stories/Table.stories.tsx:1727]: Can you pass args to these `ControllingResize` ones?

**Review by reidbarber — APPROVED** [review]
LGTM, are there any chromatic changes expected for the header cell width?

**Comment by snowystinger** [comment]
> LGTM, are there any chromatic changes expected for the header cell width?

good call, yes, there should be. I'll run it now

**Comment by snowystinger** [comment]
done, it's green

File: packages/@react-spectrum/table/stories/Table.stories.tsx [hunk 3]
```diff
@@ -28,38 +29,136 @@ import {HidingColumns} from './HidingColumns';
...
+  },
+  argTypes: {
+    // intentionally added so that we can unset the default value
+    // there is no argType for function
+    // use the controls reset button to undo it
+    // https://storybook.js.org/docs/react/essentials/controls#annotation
+    onAction: {
+      control: 'select',
+      options: [undefined]
+    },
```
Comment (LFDanLu) [inline packages/@react-spectrum/table/stories/Table.stories.tsx:62]: fancy, never thought to try this before

**Review by LFDanLu — APPROVED** [review]
LGTM, might wanna update the PR title so its more immediately obvious that this also contains a fix to focus ring cut offs in columns with hidden headers (just thinking about release notes haha)

File: packages/@react-spectrum/table/stories/Table.stories.tsx (hunk 3 above)
Comment (snowystinger) [reply packages/@react-spectrum/table/stories/Table.stories.tsx:62]: it took a while to figure this one out, it's not ideal

**Pull request merged**